python/tank/bootstrap/resolver.py

```diff
@@ -300,47 +309,30 @@ def _get_pipeline_configurations_for_project(self, pipeline_config_name, current
             "The following pipeline configurations were found: %s" % pprint.pformat(pipeline_configs)
         )
 
+        # loop over all pipeline configs
         for pipeline_config in pipeline_configs:
-            # We'll need to provide a descriptor object for the config if
-            # possible. Note that it's possible that we'll be returning a
-            # None for the config descriptor. It's up to other filtering
-            # operations to remove those if desired.
-            #
 
-            path = ShotgunPath.from_shotgun_dict(pipeline_config)
-            # If we have a plugin based pipeline.
-            if (
-                self._match_plugin_id(pipeline_config.get("plugin_ids")) or
-                self._match_plugin_id(pipeline_config.get("sg_plugin_ids"))
-            ):
-
-                # If a location was specified to get access to that pipeline, return it. Note that
-                # we are potentially returning pipeline configurations that have been configured for
-                # one platform but not all.
-                if pipeline_config.get("descriptor") or pipeline_config.get("sg_descriptor") or path:
-                    # Create a descriptor only if the pipeline is valid.
+            # see if the pipeline configuration we are looking at is relevant. Either of:
+            # - Be a match against the resolver's associated plugin id
+            # - Be a classic config associated with the resolver's associated project
+
+            if self._matches_current_plugin_id(pipeline_config) or \
+                    self._is_classic_pc_for_current_project(pipeline_config):
+
+                # extract the location information and place in special 'config_descriptor'
+                # field. Note that this may be None if for example the pipeline configuration
+                # is defined for another operating system.
+                try:
                     pipeline_config["config_descriptor"] = self._create_config_descriptor(
                         sg_connection, pipeline_config
                     )
                     yield pipeline_config
-                else:
+
+                except TankBootstrapInvalidPipelineConfigurationError:
```

Comment:
by raising here the logic can be untangled a bit and some edge cases which weren't previously not encountered for.